### PR TITLE
85-Interactive-Tutorial

### LINKED ISSUE #85: Interactive Tutorial for Getting Started
## Describe the feature you want the app to have that it currently doesn't
* Create an [interactive tutorial](https://www.swift.org/documentation/docc/tutorial-syntax) for getting started with project, this should include how to set up the project using the commands in the [README](https://github.com/mikaelacaron/Basic-Car-Maintenance#readme) and [CONTRIBUTING](https://github.com/mikaelacaron/Basic-Car-Maintenance/blob/main/CONTRIBUTING.md) file

## Describe alternatives you've considered
* N/A

## Additional context
* This issue is only about the "getting started" what a contributor would look at if they were getting started to build the project for the first time

### PR DESCRIPTION
**Pull request by ampsonic**
# What it Does
* Closes #85 
* Creates an interactive tutorial for how to get started

# Notes
* In order to view the documentation, in Xcode select Product>Build Documentation

# Screenshot
CleanShot 2023-10-17 at 12 40 33@2x

### CODE REVIEW

**Comment by mikaelacaron** [comment]
I haven't looked at this yet, but can you make dark screenshots too for GitHub? @ampsonic

**Comment by ampsonic** [comment]
> I haven't looked at this yet, but can you make dark screenshots too for GitHub? @ampsonic

You would like the screenshots inside the tutorial to be in dark mode, is that correct?

**Comment by mikaelacaron** [comment]
Almost! You can have light _and_ dark mode screenshots! DocC figures out which one to use depending on your computer settings

This [WWDC video](https://developer.apple.com/wwdc23/10244) shows you how

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 1]
```diff
@@ -0,0 +1,103 @@
+@Tutorial(time: 15) {
+    @Intro(title: "How to contribute to this project.") {
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]: Capitalize this title please

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            @Step {
+                Read the Code of Conduct
+            }
+            @Step {
+                Read the CONTRIBUTING guidelines
+            }
+            @Step {
+                Download Xcode 15 or later
+            }
+            @Step {
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial:26]: Can you add a link to this?

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                Clone your fork to your computer. 
+                
+                gh repo clone USERNAME/Basic-Car-Maintenance
+                
+                @Image(source: 3-clone.png, alt: "CLI github clone")
+            }
+        }
+    }
+    @Section(title: "Configure and Build") {
+        @ContentAndMedia {
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial:62]: Can you edit the description to: "Configuring the project with the xcconfig files to build the project on your machine"

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                Build project in Xcode
+
+                @Image(source: 6-build.png, alt: "Basic Car Maitnenace running in simulator")
+            }
+        }
+    }
+    
+    @Section(title: "Starting on an issue") {
+        @ContentAndMedia {
+            BEFORE starting on an issue, comment on the issue you want to work on.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]: can you make the work before also bold?

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+            This prevents two people from working on the same issue. Mikaela (the maintainer) will assign you that issue, and you can get started on it.
+
+            @Image(source: 7-issue.png, alt: "GitHub issue")
+        }
+        
+        @Steps {
+            @Step {
+                Checkout a new branch (from the dev branch) to work on an issue. The feature-name part of the branch can be shortened or omitted and you add your username instead. No commits should be made to the main branch directly. The main branch shall only consist of the deployed code. Developers are expected to work on feature branches, and upon successful development and testing, a PR (pull request) must be opened to merge with dev
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]: The first sentence I'd put in the "step" the rest of it, I'd add as a "note"

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                
+                git checkout -b issueNumber-feature-name
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]: Is it possible to make all the commands "formatted as code?" as opposed to it just looking like text?

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
Great work!! A couple of questions on some of the formatting.

Also with Terminal, can you install `https://ohmyz.sh/` onto your machine? The default theme will change the look of your terminal so that it doesn't have your name in all of the terminal screenshots, and instead would look something like this

Screenshot 2023-10-17 at 10 18 26 PM

If you have a question on these changes, you can comment on the thread, otherwise if you don't have any questions, click resolve, and you've addressed everything please click the re-review button
Re-review button

**Comment by mikaelacaron** [comment]
@ampsonic actually don't worry about the light / dark mode right now, I'll add that as an issue to do later

Can you fix the other comments, and then I'll be able to merge this!

**Comment by ampsonic** [comment]
> @ampsonic actually don't worry about the light / dark mode right now, I'll add that as an issue to do later
> 
> Can you fix the other comments, and then I'll be able to merge this!

Sounds good, on it!

**Comment by mikaelacaron** [comment]
@ampsonic One thing though, can you please still change the terminal stuff (like I mentioned with ohmyz.sh) and on the images, they show your profile picture, can you hide that with a circle on top of it, so it's not confusing to others?

Otherwise another thing, if you want, I can make these screenshots for you, and you add them into the project. Let me know which

**Comment by ampsonic** [comment]
> @ampsonic One thing though, can you please still change the terminal stuff (like I mentioned with ohmyz.sh) and on the images, they show your profile picture, can you hide that with a circle on top of it, so it's not confusing to others?
> 
> Otherwise another thing, if you want, I can make these screenshots for you, and you add them into the project. Let me know which

I think I already have the screenshots updated with OhMyZSH, unless I made a mistake when I pushed.   But I can add a circle to the GH screenshots.

**Comment by mikaelacaron** [comment]
> I think I already have the screenshots updated with OhMyZSH

I'll take a look later! I haven't checked back yet, but I wanted to note those I did want to change, colors are fine, I'll make a task to update them later

Thanks for updating this!

**Comment by ampsonic** [comment]
> > I think I already have the screenshots updated with OhMyZSH
> 
> I'll take a look later! I haven't checked back yet, but I wanted to note those I did want to change, colors are fine, I'll make a task to update them later
> 
> Thanks for updating this!

Committed the updated GitHub screenshots with my avatar replaced.

File: Basic-Car-Maintenance/Documentation.docc/Tutorial Table of Contents.tutorial [hunk 7]
```diff
@@ -0,0 +1,15 @@
+@Tutorials(name: "Basic-Car-Maitenance") {
+    @Intro(title: "Getting Started") {
+        How to get started contributing to this project.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/Tutorial Table of Contents.tutorial]:
```suggestion
        How to get started contributing to this project
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        
+        @Steps {
+            @Step {
+                Read the Code of Conduct
+            }
+            @Step {
+                Read the CONTRIBUTING guidelines
+            }
+            @Step {
+                Download [Xcode 15](https://developer.apple.com/xcode/resources/) or later.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
                Download [Xcode 15.0](https://developer.apple.com/xcode/resources/) or later.
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 9]
```diff
@@ -0,0 +1,106 @@
+@Tutorial(time: 15) {
+    @Intro(title: "How to Contribute to this Project.") {
+        Here you will find instructions on how to prepare, fork, clone, configure, build and contribute to this project.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
        Here's instructions on how to prepare, fork, clone, configure, build and contribute to this project. This tutorial uses the [GitHub CLI](https://cli.github.com/)
```

Rephrasing, also linking that this specifically uses the GitHub CLI

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                Read the CONTRIBUTING guidelines
+            }
+            @Step {
+                Download [Xcode 15](https://developer.apple.com/xcode/resources/) or later.
+                
+            }
+            @Step {
+                Install Github command line tools
+            }
+
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
            @Step {
                Install [GitHub CLI](https://cli.github.com/) command line tools
            }

            @Step {
               Use `gh auth login` to authenticate with the [GitHub CLI](https://cli.github.com/manual/gh_auth_login)
            }
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+                
+                `cp Basic-Car-Maintenance.xcconfig.template Basic-Car-Maintenance.xcconfig`
+                
+                @Image(source: 4-config.png, alt: "Copying template file")
+            }
+            
+            @Step {
+                In the Basic-Car-Maintenance.xcconfig file, fill in your DEVELOPMENT_TEAM and PRODUCT_BUNDLE_IDENTIFIER. You can find this by logging into the Apple Developer Portal
+                
+                This works with both free or paid Apple Developer accounts. Do NOT run this app on a real device due to issues with the Sign in With Apple capability.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
                Note: A free account does not have a `DEVELOPMENT_TEAM`. Instead fill in `ABC123`. Do NOT run this app on a real device due to issues with the Sign in With Apple capability.
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+                @Image(source: 6-build.png, alt: "Basic Car Maitnenace running in simulator")
+            }
+        }
+    }
+    
+    @Section(title: "Starting on an issue") {
+        @ContentAndMedia {
+            **BEFORE** starting on an issue, comment on the issue you want to work on.
+
+            This prevents two people from working on the same issue. Mikaela (the maintainer) will assign you that issue, and you can get started on it.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
            This prevents two people from working on the same issue. [Mikaela](https://github.com/mikaelacaron) (the maintainer) will assign you that issue, and you can get started on it.
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+            @Image(source: 7-issue.png, alt: "GitHub issue")
+        }
+        
+        @Steps {
+            @Step {
+                Checkout a new branch (from the dev branch) to work on an issue.
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
                Checkout a new branch (from the `dev` branch) to work on an issue. 
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+            This prevents two people from working on the same issue. Mikaela (the maintainer) will assign you that issue, and you can get started on it.
+
+            @Image(source: 7-issue.png, alt: "GitHub issue")
+        }
+        
+        @Steps {
+            @Step {
+                Checkout a new branch (from the dev branch) to work on an issue. 
+                
+                The feature-name part of the branch can be shortened or omitted and you add your username instead. No commits should be made to the main branch directly. The main branch shall only consist of the deployed code. Developers are expected to work on feature branches, and upon successful development and testing, a PR (pull request) must be opened to merge with dev. `git checkout -b issueNumber-feature-name`
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
                The `feature-name` part of the branch can be shortened or omitted and you add your username instead. No commits should be made to the `main` branch directly. The `main` branch shall only consist of the deployed code. Developers are expected to work on feature branches, and upon successful development and testing, a PR (pull request) must be opened to merge with `dev`. `git checkout -b issueNumber-feature-name`
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+            
+            @Image(source: 2-fork.png, alt: "GitHub Fork")
+        }
+        
+        @Steps {
+            @Step {
+                Fork the repo on GitHub to your own account.  Select Fork from the Basic-Car-Maintenance github page.
+                
+                @Image(source: 2-fork.png, alt: "GitHub Fork")
+            }
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
            @Step {
                Fork the repo on GitHub to your own account.  Select Fork from the Basic-Car-Maintenance github page.
                
                This example shows a fork to `ampsonic`'s profile
                
                @Image(source: 2-fork.png, alt: "GitHub Fork")
            }
```

File: Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial [hunk 16]
```diff
@@ -0,0 +1,106 @@
+@Tutorial(time: 15) {
+    @Intro(title: "How to Contribute to this Project.") {
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial]:
```suggestion
    @Intro(title: "How to Contribute to this Project") {
```

**Review by mikaelacaron — APPROVED** [review]
Great work!! Thanks for making those changes, I made small edits related to formatting and some phrasing

**Pull request merged**